common/sessionParsing.ts

```diff
@@ -237,6 +237,34 @@ export function parseToolCallDetails(
 				} : undefined
 			};
 		}
+	} else if (name === 'str_replace') {
+		const filePath = args.path;
+		const fileLabel = filePath && toFileLabel(filePath);
+
+		return {
+			toolName: 'Edit',
+			invocationMessage: fileLabel ? `Edit [](${fileLabel})` : `Edit ${filePath}`,
+			pastTenseMessage: fileLabel ? `Edit [](${fileLabel})` : `Edit ${filePath}`,
+			toolSpecificData: fileLabel ? {
+				command: 'str_replace',
+				filePath: filePath,
+				fileLabel: fileLabel,
+			} : undefined
+		}
+	} else if (name === 'create') {
```

Comment:
This if statement is getting huge! It would be great if we could clean it up a bit during debt week.